katello/katello-restore

```diff
@@ -31,39 +42,111 @@ end
 def reset_katello
   puts "Resetting Katello"
   `tar --selinux --overwrite --listed-incremental=/dev/null -xzf config_files.tar.gz -C /`
-  `foreman-installer --scenario katello --reset`
+  installer = "yes | foreman-installer -v --reset"
+  if @is_foremnan_proxy_content
+    installer << " --scenario #{FOREMAN_PROXY_CONTENT} --foreman-proxy-register-in-foreman false"
+  else
+    installer << " --scenario katello"
+  end
+  installer << " --disable-system-checks" if @disable_system_checks
+  puts installer
+  installer_output = `#{installer}`
+  puts installer_output
+
   puts "Done.\n"
 end
 
+def restore_psql_dumps
+  puts "Restoring postgres dump files"
+  `service postgresql start`
+  `runuser - postgres -c "dropdb foreman"`
+  `runuser - postgres -c "dropdb candlepin"`
+  `runuser - postgres -c "psql -f #{DIR}/pg_globals.dump postgres 2>/dev/null"`
+  `runuser - postgres -c "pg_restore -C -d postgres #{DIR}/foreman.dump"`
+  `runuser - postgres -c "pg_restore -C -d postgres #{DIR}/candlepin.dump"`
+  `service postgresql stop`
+  puts "Done."
+end
+
+def migrate_pulp
+  puts "Migrating pulp databases"
+  necessary_services = "mongod qpidd"
+  pulp_services = "pulp_celerybeat pulp_workers pulp_resource_manager"
+  `systemctl start #{necessary_services}`
+  `systemctl stop #{pulp_services}`
```

Comment:
We bake some specialness into katello-service regarding each service for starting and stopping them. I lean towards it being safer to call it using the `only` option.